cmd/main.go

```diff
@@ -3,13 +3,19 @@ package main
 import (
 	"log"
 
-	repositories "github.com/cyneptic/letsgo/internal/infrastructure/repository"
+	"github.com/cyneptic/letsgo/controller"
+	repositories "github.com/cyneptic/letsgo/infrastructure/repository"
+	"github.com/cyneptic/letsgo/internal/core/service"
 	"github.com/labstack/echo/v4"
 )
 
 func main() {
-	_ = repositories.NewGormDatabase()
 	e := echo.New()
+	db := repositories.NewPostgres()
+	redis := repositories.RedisInit()
+	userService := service.NewUserService(db , redis)
```

Comment:
these should be moved into the respective factory function.

db and redis into NewUserService()
and userService into NewUserHandler()